src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1658,6 +1658,27 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
         {
             throw new PlatformNotSupportedException();
         }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern void* GetEnCFieldAddr(object? tgt, void* pFD);
+
+        // implementation of CORINFO_HELP_GETFIELDADDR
+        internal static unsafe void* GetFieldAddr(object tgt, void* pFD)
+        {
+            void* addr = GetEnCFieldAddr(tgt, pFD);
+            if (addr == null)
+                throw new NullReferenceException();
+            return addr;
+        }
+
+        // implementation of CORINFO_HELP_GETSTATICFIELDADDR
+        internal static unsafe void* GetStaticFieldAddr(void* pFD)
```

Comment:
```suggestion
        // implementation of CORINFO_HELP_GETFIELDADDR
        [StackTraceHidden]
        internal static unsafe void* GetFieldAddr(object tgt, void* pFD)
        {
            void* addr = GetEnCFieldAddr(tgt, pFD);
            if (addr == null)
                throw new NullReferenceException();
            return addr;
        }

        // implementation of CORINFO_HELP_GETSTATICFIELDADDR
        [StackTraceHidden]
        internal static unsafe void* GetStaticFieldAddr(void* pFD)
```

HCalls are hidden from stack traces.

Reply:
We typically end up marking the JIT helpers with:

```
    [StackTraceHidden]
    [DebuggerStepThrough]
    [DebuggerHidden]
```